File: Module - Speech/Speech/language/src/main/java/com/javadude/speech/Room.java
package com.javadude.speech;

import java.util.HashMap;
import java.util.Map;

public class Room extends ItemHolder {
	private Map<Direction, Room> paths = new HashMap<Direction, Room>();
	
	public Room(String name, String description) {
		super(name, description, false);
	}
	public String getDescription() {
		String result = super.getDescription();
		if (!paths.isEmpty()) {
			result += "\n\n" + "You can go ";
			for (Direction d : paths.keySet()) {
				result += d.name() + ", ";
			}
			result = result.substring(0, result.length()-2);
		}
		if (!isEmpty()) {
			result += "\n\n" + "You see ";
			for (String itemName : getItemNames()) {
				result += "a " + itemName + ", ";
			}
			result = result.substring(0, result.length()-2);
		}
		return result;
	}
	public Room move(Direction direction) {
		return paths.get(direction);
	}
	public void connect(Direction direction, Room room) {
		paths.put(direction, room);
	}
}

File: Module - Speech/Speech2/language/src/main/java/com/javadude/speech2/Room.kt
package com.javadude.speech2

import java.util.*

class Room(name: String,
           description: String)
        : ItemHolder(name, description, false) {

    private val paths = HashMap<Direction, Room>()
    override val description: String
        get() {
            var result = super.description
            if (paths.isNotEmpty()) {
                result += "\n\n" + "You can go "
                for (d in paths.keys) {
                    result += d.name + ", "
                }
                result = result.substring(0, result.length - 2)
            }
            if (!isEmpty) {
                result += "\n\n" + "You see "
                for (itemName in itemNames) {
                    result += "a $itemName, "
                }
                result = result.substring(0, result.length - 2)
            }
            return result
        }

    fun move(direction: Direction) = paths[direction]

    fun connect(direction: Direction, room: Room) {
        paths[direction] = room
    }
}
